### ISSUE TITLE
Audio device operations incomplete

### ISSUE THREAD

**Issue by dstengle** [kunitoki/popsicle#27]
I'm trying to get an audio device and the device manager gives me None for the current device. I notice there are a number of methods missing regarding audio devices. Is audio device functionality not currently developed?

**Comment by kunitoki** [comment]
Which version are you using ? Current master is based on juce8 and it is not tested much.

Does this work for you ? https://github.com/kunitoki/popsicle/blob/master/examples/audio_device.py

**Comment by dstengle** [comment]
I get this error:

`
Traceback (most recent call last):
  File "/home/dstengle/Downloads/audio_device.py", line 4, in <module>
    from juce_init import START_JUCE_COMPONENT
ModuleNotFoundError: No module named 'juce_init'
`
I'm on 0.9.6 through pypi. Should I downgrade?

**Comment by kunitoki** [comment]
juce_init is part of the examples, you need to run from the examples folder

**Comment by dstengle** [comment]
I had figured that out and then forgot. This is what I get:

`python audio_device.py 
no channels`

Do I need to set any env variables? I'm on ubuntu 24.04.

**Comment by kunitoki** [comment]
It seems like juce is not able to find an Alsa device, not sure why.

**Comment by dstengle** [comment]
You know, I had worked that out before and forgot! I opened my shell in there and I had already run the example:

"no channels"

Do I need to set up any env variables

**Comment by kunitoki** [comment]
I don't think so, but i would check if you can open a device with 2 output channels, maybe your soundcard don't allow it (you can only open 1 or 4 for examples) but you should try to get some more info from the AudioDeviceManager
